apps/src/templates/instructions/TopInstructionsCSP.jsx

```diff
@@ -414,61 +459,110 @@ class TopInstructionsCSP extends Component {
+                isMinecraft={this.props.isMinecraft}
               />
-              {displayHelpTab && (
+              {this.props.noInstructionsWhenCollapsed && displayHelpTab && (
                 <InstructionsTab
                   className="uitest-helpTab"
                   onClick={this.handleHelpTabClick}
                   selected={this.state.tabSelected === TabType.RESOURCES}
-                  text={msg.helpTips()}
+                  text={i18n.helpTips()}
                   teacherOnly={teacherOnly}
+                  isMinecraft={this.props.isMinecraft}
                 />
               )}
-              {displayFeedback && (!this.state.fetchingData || teacherOnly) && (
-                <InstructionsTab
-                  className="uitest-feedback"
-                  onClick={this.handleCommentTabClick}
-                  selected={this.state.tabSelected === TabType.COMMENTS}
-                  text={feedbackTabText}
+              {this.props.noInstructionsWhenCollapsed &&
+                displayFeedback &&
+                (!this.state.fetchingData || teacherOnly) && (
+                  <InstructionsTab
+                    className="uitest-feedback"
+                    onClick={this.handleCommentTabClick}
+                    selected={this.state.tabSelected === TabType.COMMENTS}
+                    text={feedbackTabText}
+                    teacherOnly={teacherOnly}
+                    isMinecraft={this.props.isMinecraft}
+                  />
+                )}
+              {!this.props.noInstructionsWhenCollapsed &&
+                this.props.viewAs === ViewType.Teacher &&
+                this.props.teacherMarkdown && (
+                  <InstructionsTab
+                    className="uitest-teacherOnlyTab"
+                    onClick={this.handleTeacherOnlyTabClick}
+                    selected={this.state.tabSelected === TabType.TEACHER_ONLY}
+                    text={i18n.teacherOnly()}
+                    teacherOnly={teacherOnly}
+                    isMinecraft={this.props.isMinecraft}
+                  />
+                )}
+            </div>
+            {!this.props.isEmbedView &&
+              (this.props.noInstructionsWhenCollapsed ||
+                this.props.hasContainedLevels) && (
+                <CollapserIcon
+                  collapsed={this.props.collapsed}
+                  onClick={this.handleClickCollapser}
                   teacherOnly={teacherOnly}
                 />
               )}
-            </div>
-            {!this.props.isEmbedView && (
-              <CollapserIcon
-                collapsed={this.props.collapsed}
-                onClick={this.handleClickCollapser}
-                teacherOnly={teacherOnly}
-              />
-            )}
           </div>
         </PaneHeader>
-        <div style={[this.props.collapsed && commonStyles.hidden]}>
-          <div style={styles.body}>
+        <div
+          style={[
```

Comment:
this `style` prop can be an object instead of an array since we are only ever applying 1 style at most. we usually use styles arrays when there are 1+ style objects being applied (like on lines 557-564 below). something like this should work:

```
style={(this.props.collapsed && this.props.noInstructionsWhenCollapsed) ? commonStyles.hidden : {}}
```

or i think this would work as well:
```
style={this.props.collapsed && this.props.noInstructionsWhenCollapsed && commonStyles.hidden}
```

Reply:
It gives an error every time I try to take this away ``` invariant.js:42 Uncaught Invariant Violation: The `style` prop expects a mapping from style properties to values, not a string. For example, style={{marginRight: spacing + 'em'}} when using JSX. This DOM node was rendered by `TopInstructionsCSP`. ```

Reply:
strange, i'm not sure i understand why. that's okay though -- not a super important refactor to make